cst/plugins/eu.esdihumboldt.cst.functions.collector/src/eu/esdihumboldt/cst/functions/collector/explanations/AssignFromCollectorExplanation.java

```diff
@@ -41,38 +40,32 @@
 	protected void customizeBinding(Map<String, Object> binding, Cell cell, boolean html,
 			ServiceProvider provider, Locale locale) {
 
-		FunctionDefinition<? extends ParameterDefinition> function = loadFunction(
-				cell.getTransformationIdentifier(), provider);
-
 		// Default values for all bindings
 		binding.put("_constraintsEvaluated", false);
 		binding.put("_hasValue", false);
 		binding.put("_isReference", false);
 
-		if (!function.getTarget().isEmpty() && !cell.getTarget().isEmpty()) {
-
-			ParameterDefinition parameterDefinition = function.getTarget().iterator().next();
-			Entity entity = cell.getTarget().get(parameterDefinition.getName()).iterator().next();
-			if (entity != null) {
-				ChildContext childContext = entity.getDefinition().getPropertyPath().iterator()
-						.next();
-				PropertyDefinition resultProperty = childContext.getChild().asProperty();
-				TypeDefinition resultPropertyType = resultProperty.getPropertyType();
-
-				boolean isReference = resultProperty.getConstraint(Reference.class).isReference();
-				binding.put("_isReference", isReference);
+		Entity entity = CellUtil.getFirstEntity(cell.getTarget());
+		if (entity != null) {
+			ChildContext childContext = entity.getDefinition().getPropertyPath().iterator().next();
```

Comment:
That would only be the first element in the property path, and not necessarily the actual property that is mapped (could be a parent property). The best way is to do `entity.getDefinition().getDefinition()` and use that - that is always the last element of the path. You can just check if it is a `PropertyDefinition` and cast it accordingly.